src/classes/BDI_SettingsUI_CTRL.cls

```diff
@@ -133,45 +125,36 @@ public with sharing class BDI_SettingsUI_CTRL {
     public list<SelectOption> listSOAccountUniqueID {
         get {
             if (listSOAccountUniqueID == null) {
-                listSOAccountUniqueID = new list<SelectOption>();
-                //add a default option
-                listSOAccountUniqueID.add(new SelectOption('', Label.stgLabelNone));
-                                           
-                Map<String, Schema.DescribeFieldResult> mapNameToDfr = UTIL_Describe.getAllFieldsDescribe('Account');               
-                for (Schema.DescribeFieldResult dfr : mapNameToDfr.values()) {
-                    //only allow custom fields that have no namespace
-                    if (dfr.isCustom() && dfr.getName().countMatches('__') == 1) {
-                        Schema.Displaytype dt = dfr.getType();
-                        // only strings and numbers
-                        if (dt == Schema.DisplayType.String ||
-                           dt == Schema.DisplayType.Integer ||
-                           dt == Schema.DisplayType.Double) {
-                           listSOAccountUniqueID.add(new SelectOption(dfr.getName(), dfr.getLabel()));
-                        } 
-                    }               
-                }
-            }               
+                listSOAccountUniqueID = listSOForObject('Account');
+            }
             return listSOAccountUniqueID;
         }
         private set;
     }
 
     /*********************************************************************************************************
-    * @description saves the provided Settings.  since we want this operation controlled by the containing
-    * page, it is static to be callable by the controlling page.  it must pass the settings instance however.
-    * @param diSettings The BDI settings to save
-    * @return True if success, False if error.
+    * @description creates a list of SelectOptions of potential Unique ID fields from the specified object
+    * @param strObject the object to query fields on
+    * @return list<SelectOption>
     */
-    public static boolean saveBDISettings(Data_Import_Settings__c diSettings) {
+    private list<SelectOption> listSOForObject(String strObject) {
```

Comment:
I take it you're not using dfr.isExternalID() in case they're using a field that's not marked as external ID?

I'm also wondering if it's possible that a unique ID field could be packaged at some point, are you sure showing only unpackaged fields makes sense?

Reply:
yes, the field does not have to be an official external ID.  As for packaged fields from another package, I entered issue #2700 several weeks ago that BDI does not correctly support them.